File: rb/features.rb
# YaglaUrlShortener SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module YaglaUrlShortenerFeatures
  def self.make_feature(name)
    case name
    when "base"
      YaglaUrlShortenerBaseFeature.new
    when "test"
      YaglaUrlShortenerTestFeature.new
    else
      YaglaUrlShortenerBaseFeature.new
    end
  end
end

File: py/features.py
# YaglaUrlShortener SDK feature factory

from feature.base_feature import YaglaUrlShortenerBaseFeature
from feature.test_feature import YaglaUrlShortenerTestFeature


def _make_feature(name):
    features = {
        "base": lambda: YaglaUrlShortenerBaseFeature(),
        "test": lambda: YaglaUrlShortenerTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
